File: seminar_2/task_2/patokova.victoria/task-2.py
def is_palindrome(s):
    stack = list()

    for char in s:
        stack.push(char)

    for char in s:
        if char != stack.pop():
            return False

    return True


def is_palindrome_dpointers(s):
    l = 0
    r = len(s) - 1

    while l <= r:
        if s[l] != s[r]:
            return False
        l += 1
        r -= 1

    return True

File: seminar_2/task_2/churilkin.artem/task2.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;    

bool isPalindrome(string word) {
    stack<char> s;
    int len = word.length();

    for (int i = 0; i < len; ++i) {
        s.push(word[i]);
    }

    for (int i = 0; i < len; ++i) {
        if (word[i] != s.top()) {
            return false;
        }
        s.pop();
    }

    return true;
}

int main() {
    string word;
    cin >> word;

    if (isPalindrome(word)) {
        cout << "True";
    } else {
        cout << "False";
    }

    return 0;
}
